### PR TITLE
Improve demos

### LINKED ISSUE #155: Improving vaadin checkbox demos

### PR DESCRIPTION
**Pull request by shadikhani**
Fix #155

### CODE REVIEW

File: demo/checkbox-basic-demos.html [hunk 1]
```diff
@@ -6,36 +6,131 @@
...
+    <vaadin-demo-snippet>
       <template preserve-content>
-        <vaadin-checkbox checked>
-          <iron-icon icon="lumo:bell" style="vertical-align: top;"></iron-icon>
-          Notifications
-          <br>
-          <small>Enable desktop notifications</small>
-        </vaadin-checkbox>
+        <vaadin-checkbox-group label="Label" theme="vertical">
+          <vaadin-checkbox value="Option one" checked>Option one</vaadin-checkbox>
```
Comment (tomivirkki) [inline demo/checkbox-basic-demos.html]: The values should be something else than the formatted/localized labels. Maybe "1", "2" and "3".

Comment (shadikhani) [reply demo/checkbox-basic-demos.html]: Done.

File: demo/checkbox-presentation-demos.html [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        <img style="width: 21px; border-radius: 50%; vertical-align: bottom;"
+          src="https://randomuser.me/api/portraits/women/43.jpg" />
+        Gabriella
+      </vaadin-checkbox>
+      <vaadin-checkbox value="2">
+        <img style="width: 21px; border-radius: 50%; vertical-align: bottom;"
+          src="https://randomuser.me/api/portraits/men/77.jpg" />
+        Rudi
+      </vaadin-checkbox>
+      <vaadin-checkbox value="Hamsa">
```
Comment (tomivirkki) [inline demo/checkbox-presentation-demos.html]: This `value="Hamsa"` intentional?

Comment (shadikhani) [reply demo/checkbox-presentation-demos.html]: Done.

Comment (shadikhani) [reply demo/checkbox-presentation-demos.html]: Done.

File: demo/checkbox-validation-demos.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        <vaadin-checkbox-group label="Required" required error-message="Please choose one option" theme="vertical"
+          id="checkboxGroup">
+          <vaadin-checkbox value="1">Option one</vaadin-checkbox>
+          <vaadin-checkbox value="2">Option two</vaadin-checkbox>
+          <vaadin-checkbox value="3">Option three</vaadin-checkbox>
+        </vaadin-checkbox-group>
+        <vaadin-button id="submit">Submit</vaadin-button>
+        <script>
+          window.addDemoReadyListener('#checkbox-required', function(document) {
+            var submit = document.getElementById('submit');
```
Comment (tomivirkki) [inline demo/checkbox-validation-demos.html]: nit: We could align these better with the other demos by changing:
`var` -> `const`
`getElementById` -> `querySelector`

File: demo/checkbox-basic-demos.html [hunk 4]
```diff
@@ -6,36 +6,131 @@
...
+          <vaadin-checkbox value="2">Option two</vaadin-checkbox>
+          <vaadin-checkbox value="3">Option three</vaadin-checkbox>
+        </vaadin-checkbox-group>
+        <div id="output">Selected value:</div>
+        <script>
+          window.addDemoReadyListener('#value-change-event', function(document) {
+            const checkboxGroup = document.querySelector('vaadin-checkbox-group');
+            const output = document.querySelector('#output');
+            checkboxGroup.addEventListener('change', function(event) {
+              output.textContent = checkboxGroup.value;
```
Comment (tomivirkki) [inline demo/checkbox-basic-demos.html]: The output behaves a bit stange. First it says `Selected value:` but when you select items, it clears the text and just says `1,2`

File: demo/checkbox-presentation-demos.html [hunk 5]
```diff
@@ -0,0 +1,68 @@
+<dom-module id="checkbox-presentation-demos">
+  <template>
+    <style include="vaadin-component-demo-shared-styles">
+      :host {
+        display: block;
+      }
+    </style>
+
+<h3>Customizing checkbox label</h3>
```
Comment (tomivirkki) [inline demo/checkbox-presentation-demos.html]: Indentation is reset on this line

File: demo/checkbox-validation-demos.html (hunk 3 above)
Comment (shadikhani) [reply demo/checkbox-validation-demos.html]: Done.

File: demo/checkbox-basic-demos.html (hunk 4 above)
Comment (shadikhani) [reply demo/checkbox-basic-demos.html]: Done.

File: demo/checkbox-presentation-demos.html (hunk 5 above)
Comment (shadikhani) [reply demo/checkbox-presentation-demos.html]: Done.

File: demo/checkbox-basic-demos.html (hunk 4 above)
Comment (tomivirkki) [reply demo/checkbox-basic-demos.html]: It still works the same as before:
- First it says `Selected value:` (followed with a ":", indicating the selected values will be listed after it)
- Selecting values still replaces the "Selected value:" text with "Option two,Option three". I think the previous values "1,2" were better, but what it should say is "Selected value: 1, 2" (not clearing the "Selected value:" text).

Maybe it makes sense to have a separate container for the value. Something like `<div>Selected value: <span id="output"></span></div>`

Comment (shadikhani) [reply demo/checkbox-basic-demos.html]: Done.

**Pull request merged**